web_timeline/static/src/js/timeline_renderer.js

```diff
@@ -0,0 +1,536 @@
+            if (mode && adjust) {
+                this.timeline.fit();
+            }
+        },
+
+        /**
+         * Get the groups.
+         *
+         * @private
+         * @returns {Array}
+         */
+        split_groups: function (events, group_bys) {
+            if (group_bys.length === 0) {
+                return events;
+            }
+            var groups = [];
+            groups.push({id: -1, content: _t('-')});
+            _.each(events, function (event) {
+                var group_name = event[_.first(group_bys)];
+                if (group_name) {
+                    if (group_name instanceof Array) {
+                        var group = _.find(groups, function (existing_group) {
+                            return _.isEqual(existing_group.id, group_name[0]);
+                        });
+
+                        if (_.isUndefined(group)) {
+                            group = {
+                                id: group_name[0],
+                                content: group_name[1],
+                            };
+                            groups.push(group);
+                        }
+                    }
+                }
+            });
+            return groups;
+        },
+
+        /**
+         * Transform Odoo event object to timeline event object.
+         *
+         * @private
+         * @returns {Object}
+         */
+        event_data_transform: function (evt) {
+            var self = this;
+            var date_start = new moment();
+            var date_stop = null;
+
+            var date_delay = evt[this.date_delay] || false,
+                all_day = this.all_day ? evt[this.all_day] : false;
+
+            if (all_day) {
+                date_start = time.auto_str_to_date(evt[this.date_start].split(' ')[0], 'start');
+                if (this.no_period) {
+                    date_stop = date_start;
+                } else {
+                    date_stop = this.date_stop ? time.auto_str_to_date(evt[this.date_stop].split(' ')[0], 'stop') : null;
+                }
+            } else {
+                date_start = time.auto_str_to_date(evt[this.date_start]);
+                date_stop = this.date_stop ? time.auto_str_to_date(evt[this.date_stop]) : null;
+            }
+
+            if (!date_stop && date_delay) {
+                date_stop = moment(date_start).add(date_delay, 'hours').toDate();
+            }
+
+            var group = evt[self.last_group_bys[0]];
+            if (group && group instanceof Array) {
+                group = _.first(group);
+            } else {
+                group = -1;
+            }
+            _.each(self.colors, function (color) {
+                if (eval("'" + evt[color.field] + "' " + color.opt + " '" + color.value + "'")) {
+                    self.color = color.color;
+                }
+            });
+
+            var content = _.isUndefined(evt.__name) ? evt.display_name : evt.__name;
```

Comment:
I think that here can use short-circuit evaluation...
```suggestion
            const content = evt.__name || evt.display_name;
```